.github/workflows/ci.yml

```diff
@@ -0,0 +1,134 @@
+name: CI
+
+# Based on: https://github.com/input-output-hk/ouroboros-consensus/blob/11b8be10bd14ee1b8294a89dbc23b0a1e48bcc86/.github/workflows/ci.yml
+
+# Limit concurrent runs of this workflow within a single PR
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  pull_request:
+  merge_group:
+  push:
+    branches:
+      - main
+jobs:
+  build-test-bench-haddocks:
```

Comment:
```suggestion
  build:
```

Reply:
Ups. I replaced this with `build-test`. We might add tests soon.